solidity/test/StandardPoolConverter.js

```diff
@@ -1188,4 +1196,212 @@ describe('StandardPoolConverter', () => {
+                            }
+
+                            for (const n of [180, 170, 160, 150]) {
+                                const conversion = await convert(reserveToken1, poolToken, reserveToken2, bancorNetwork, converter, ONE_TOKEN.muln(n));
+                                totalConversionFee2 = totalConversionFee2.add(conversion.fee);
+                            }
+
+                            for (const n of [140, 130, 120, 110]) {
+                                const conversion = await convert(reserveToken2, poolToken, reserveToken1, bancorNetwork, converter, ONE_TOKEN.muln(n));
+                                totalConversionFee1 = totalConversionFee1.add(conversion.fee);
+                            }
+
+                            const totalSupply = await poolToken.totalSupply();
+                            const reserveBalance1 = await reserveToken1.balanceOf(converter.address);
+                            const reserveBalance2 = await reserveToken2.balanceOf(converter.address);
+
+                            const balance1Before = await reserveToken1.balanceOf(NETWORK_FEE_WALLET);
+                            const balance2Before = await reserveToken2.balanceOf(NETWORK_FEE_WALLET);
+
+                            const supplyAmount = await poolToken.balanceOf(defaultSender);
+                            await removeLiquidity(reserveToken1, reserveToken2, converter, supplyAmount, true);
+
+                            const balance1After = await reserveToken1.balanceOf(NETWORK_FEE_WALLET);
+                            const balance2After = await reserveToken2.balanceOf(NETWORK_FEE_WALLET);
+
+                            const totalConversionFee1InPoolTokenUnits = totalConversionFee1.mul(totalSupply).div(reserveBalance1);
+                            const totalConversionFee2InPoolTokenUnits = totalConversionFee2.mul(totalSupply).div(reserveBalance2);
+                            const totalConversionFeeInPoolTokenUnits = totalConversionFee1InPoolTokenUnits.add(totalConversionFee2InPoolTokenUnits);
+                            const expectedFeeBase = totalConversionFeeInPoolTokenUnits.divn(2).muln(networkFeePercent).divn(100 + networkFeePercent);
+                            const expectedFee1 = expectedFeeBase.mul(reserveBalance1).div(totalSupply);
+                            const expectedFee2 = expectedFeeBase.mul(reserveBalance2).div(totalSupply);
+
+                            const actualFee1 = balance1After.sub(balance1Before);
+                            const actualFee2 = balance2After.sub(balance2Before);
+
+                            expectAlmostEqual(actualFee1, expectedFee1, '0', '0.001367');
+                            expectAlmostEqual(actualFee2, expectedFee2, '0', '0.001367');
+                        });
+                    }
+                }
+            }
+        }
+
+        function description(prefix, initialBalance1, initialBalance2, conversionFeePercent, networkFeePercent) {
```

Comment:
In general, these are really good tests. Can you also add the following cases and make sure that the protocol behaves correctly:
1. Simulate multiple additions/removal in the same block.
2. Partial removal of liquidity.
3. Adding liquidity after a conversion.
4. Full liquidity removal, but in incremental steps (e.g., 50% and then the remainder).
5. Adding/removing very small amounts (e.g., weis) and see that the imprecision doesn't screw the system. There were similar exploits in the space already

Reply:
Done.